### PR TITLE
allowed navigation to any object via the location.hash

### PR DESCRIPTION
**Pull request by chris-steele**
e.g:

index.html#/id/co-05
index.html#/id/c-85
index.html#/id/b-30
index.html#/id/a-10

etc

Also as requested by Ollie the offset of the navigation is taken into account in Adapt.navigateToElement.

### CODE REVIEW

File: src/core/js/adapt.js [hunk 1]
```diff
@@ -43,33 +46,34 @@ define(function(require){
...
 
+        if(offset===undefined) offset = -($('.navigation').height()+10);
+
         // Setup settings object
         var settings = (settings || {});
 
         // Removes . symbol from the selector to find the model
         var currentModelId = selector.replace(/\./g, '');
-        var currentModel = Adapt.findById(currentModelId);
+        var currentModel = Adapt[Adapt.mapById(currentModelId)].findWhere({_id: currentModelId});
```
Comment (oliverfoster) [inline src/core/js/adapt.js]: findById does Adapt[Adapt.mapById(currentModelId)].findWhere({_id: currentModelId}); so you can revert this line.

File: src/core/js/router.js [hunk 2]
```diff
@@ -55,23 +55,30 @@ define(function(require) {
         
         handleId: function(id) {
             
-            this.removeViews();
-            this.showLoading();
+            var currentModel = Adapt[Adapt.mapById(id)].findWhere({_id: id});
```
Comment (oliverfoster) [inline src/core/js/router.js]: You can use findById here also

File: src/core/js/adapt.js [hunk 3]
```diff
@@ -43,33 +46,34 @@ define(function(require){
         
     }
 
-    Adapt.navigateToElement = function(selector, settings) {
+    Adapt.navigateToElement = function(selector, settings, offset) {
         // Allows a selector to be passed in and Adapt will navigate to this element
 
+        if(offset===undefined) offset = -($('.navigation').height()+10);
```
Comment (oliverfoster) [inline src/core/js/adapt.js]: can we use settings.offset instead of a new parameter?

File: src/core/js/router.js [hunk 4]
```diff
@@ -55,23 +55,30 @@ define(function(require) {
...
+
+            if (currentModel.get('_type') == 'page' || currentModel.get('_type') == 'menu')
```
Comment (oliverfoster) [inline src/core/js/router.js]: How about a switch?

switch (currentModel.get('_type)) {
case 'page': case 'menu': 
      /* menu + page loading here */
      break;
default:
      Adapt.navigateToElement('.' + id);
}

**Comment by chris-steele** [comment]
Great thanks Ollie I'll make those changes now.

**Comment by chris-steele** [comment]
Changes made - are these OK?

File: src/core/js/adapt.js [hunk 5]
```diff
@@ -49,11 +50,13 @@ define(function(require){
         // Setup settings object
         var settings = (settings || {});
 
+        if(settings.offset===undefined) settings.offset = -($('.navigation').height()+10);
```
Comment (oliverfoster) [inline src/core/js/adapt.js]: if (settings.top === undefined) settings.top = -($('.navigation').height()+10);

File: src/core/js/adapt.js [hunk 6]
```diff
@@ -33,8 +33,9 @@ define(function(require){
         if (!settings.duration) {
             settings.duration = $.scrollTo.defaults.duration;
         }
+
         // Trigger scrollTo plugin
-        $.scrollTo(selector, settings);
+        $.scrollTo(selector, settings, {top:settings.offset, left:0});
```
Comment (oliverfoster) [inline src/core/js/adapt.js]: $.scrollTo(selector, settings);

**Comment by chris-steele** [comment]
I was just looking at this line and wondering what I was thinking. Thanks!

**Comment by oliverfoster** [comment]
try passing settings.top instead... it might be a better idea?
the you can just use the old $.scrollTo(selector, settings);

**Comment by oliverfoster** [comment]
+1

**Comment by moloko** [comment]
+1

**Comment by chris-steele** [comment]
Looks good Ollie, thanks for being a bit more thorough than I(!)

+1

**Comment by hrajotia** [comment]
+1

**Comment by oliverfoster** [comment]
+1

**Comment by moloko** [comment]
+1
cheers everyone I think this will prove very useful to have. would it make sense to have things like bookmarking hook into this as well maybe?

**Pull request merged**